src/HttpKernel/Bundle/ModuleBundleGenerator.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+/*
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2016 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao\CoreBundle\HttpKernel\Bundle;
+
+use Symfony\Component\DependencyInjection\Container;
+use Symfony\Component\Filesystem\Filesystem;
+
+/**
+ * Generates classes for Contao modules
+ *
+ * @author Andreas Schempp <https://github.com/aschempp>
+ */
+class ModuleBundleGenerator
+{
+
+    public function generateBundles($cacheDir, $rootDir, Filesystem $filesystem = null)
+    {
+        if (null === $filesystem) {
+            $filesystem = new Filesystem();
+        }
+
+        $filesystem->remove($cacheDir);
+        $filesystem->mkdir($cacheDir);
+
+        $modules = $this->getContaoModules(dirname($rootDir) . '/system/modules');
+
+        foreach ($modules as $module) {
+            $this->createBundleClass($module, $cacheDir, $rootDir);
+        }
+    }
+
+    /**
+     * Find Contao modules in system/modules that are not marked as skippable
+     *
+     * @param string $modulesDir
+     *
+     * @return array
+     */
+    private function getContaoModules($modulesDir)
+    {
+        $modules = [];
+
+        foreach (scandir($modulesDir) as $dir) {
+            if ('.' === $dir || '..' === $dir) {
+                continue;
+            }
+
```

Comment:
To prevent syntax errors and security issues, we should only allow directories with certain characters IMO, like:

```php
if (preg_match('/[^a-z0-9_-.]/i', $dir)) {
	continue;
}
```